packages/api-page-builder/src/graphql/upgrades/v5.34.0/index.ts

```diff
@@ -0,0 +1,168 @@
+import { UpgradePlugin } from "@webiny/api-upgrade";
+import WebinyError from "@webiny/error";
+import { PageBuilderContextObject, PbContext } from "~/graphql/types";
+import { Tenant } from "@webiny/api-tenancy/types";
+import { I18NContextObject } from "@webiny/api-i18n/types";
+import { Folders } from "@webiny/api-folders/types";
+
+interface CreatePageLinksParams {
+    tenant: Tenant;
+    pageBuilder: PageBuilderContextObject;
+    folders: Folders;
+    i18n: I18NContextObject;
+}
+
+interface FetchPagesAndCreateLinksParams {
+    pageBuilder: PageBuilderContextObject;
+    folders: Folders;
+}
+
+const fetchPagesAndCreateLinks = async (
+    params: FetchPagesAndCreateLinksParams
+): Promise<(string | undefined)[]> => {
+    const { pageBuilder, folders } = params;
+
+    /**
+     *  Since `listPages` is paginated, we need a recursive function to fetch pages and create links.
+     *
+     * @param acc accumulator for results.
+     * @param hasMoreItems continue the iteration or return the accumulator.
+     * @param after the cursor value to pass to `listPages`.
+     */
+    async function fetchAndCreate(
+        acc: (string | undefined)[],
+        hasMoreItems: boolean,
+        after?: string | null
+    ): Promise<(string | undefined)[]> {
+        /**
+         * No more pages to fetch, return the accumulator.
+         */
+        if (!hasMoreItems) {
+            return acc;
+        }
+
+        /**
+         * Fetch page data and meta
+         */
+        const [data, meta] = await pageBuilder.listLatestPages({ limit: 100, after });
+        const pageIds = data.map(page => page.pid);
+
+        /**
+         * Create `links` and return the `id`.
+         */
+        const linkIds = await Promise.all(
+            pageIds.map(async id => {
+                try {
+                    const link = await folders.createLink({ id, folderId: "ROOT" });
+                    return link.id;
+                } catch (e) {
+                    /**
+                     * In case the link already exists for the current page, just return and continue the process.
+                     */
+                    if (e.code === "LINK_EXISTS") {
+                        return;
+                    }
+                    throw e;
+                }
+            })
+        );
+
+        const result = [...acc, ...linkIds];
+        return await fetchAndCreate(result, meta.hasMoreItems, meta.cursor);
+    }
+
+    return await fetchAndCreate([], true);
+};
+
+const createPageLinks = async (params: CreatePageLinksParams): Promise<void> => {
+    const { tenant, pageBuilder, folders, i18n } = params;
+
+    /**
+     * Find all locales for the current tenant, so we can find all pages for each locale.
+     */
+    const [locales] = await i18n.locales.storageOperations.list({
+        where: {
+            tenant: tenant.id
+        },
+        limit: 100
```

Comment:
Why limit? If there's no reason, remove this.

Reply:
`limit` is mandatory for this operation